examples/blog/src/components/HeaderLink.astro

```diff
@@ -1,8 +1,19 @@
 ---
 export interface Props extends astroHTML.JSX.AnchorHTMLAttributes {}
 
+const removeTrailingSlash = (str: string): string => {
+  if (str.charAt(str.length - 1) === "/") {
+    return str.slice(0, -1);
+  } else {
+    return str;
+  }
+};
+
 const { href, class: className, ...props } = Astro.props;
-const isActive = href === Astro.url.pathname;
+const isActive =
+  typeof href === "string" &&
+  removeTrailingSlash(href) === removeTrailingSlash(Astro.url.pathname);
+
```

Comment:
This could be simplified significantly to just trim the trailing slash if it exists!

```suggestion
const isActive = href === Astro.url.pathname.replace(/\/$/, '');
```